### DISCUSSION TITLE
Does zlib-ng speed up pigz in zopfli mode (-11)?

### DISCUSSION THREAD

**Discussion by jfikar** [zlib-ng/zlib-ng#1754] [Q&A] [answered]
Hi,
I've tried the zlib-ng-2.2.0RC on Raspberry Pi 5 with ZLIB_COMPAT=on and LD_PRELOAD. I tested then the distro default pigz at different compression levels. On levels 1-9 I see nice speedut, but for 11 (zopfli) there is no speedup.

Is it because zopfli does not use zlib? Or do I need to build pigz with zlib-ng?

Another question: is the ZLIB_COMPAT=on influencing the performance, or not?

**Comment by Dead2** [comment, marked as answer]
zopfli is another alternative to zlib/zlib-ng, optimized heavily for providing the best compression while sacrificing speed. It does not use zlib/zlib-ng, so we cannot speed that up unfortunately.

ZLIB_COMPAT=on should not influence performance noticeably. It does do a few more chained function calls compared to the native api, but it is not something that is generally noticeable. Other options would affect perfomance, such as disabling the new compression strategies, like some distros do for their zlib-compat builds.

**Comment by jfikar** [comment]
Thanks for the explanation. Makes sense.
